lua/orgmode/org/links/handlers/_internal/headline.lua

```diff
@@ -0,0 +1,68 @@
+local utils = require('orgmode.utils')
+local Org = require('orgmode')
+local Internal = require('orgmode.org.hyperlinks.builtin.internal')
+local Id = require('orgmode.org.hyperlinks.builtin.id')
+
+---@class OrgLinkHandlerHeadline:OrgLinkHandlerInternal
+local Headline = Internal:new()
+
+function Headline:new(headline)
+  ---@class OrgLinkHandlerHeadline
+  local this = Internal:new()
+  this.headline = headline
+  setmetatable(this, self)
+  self.__index = self
+  return this
+end
+
+---@param input string
+function Headline.parse(input)
+  return Headline:new(input)
+end
+
+function Headline:__tostring()
+  return string.format('*%s', self.headline)
+end
+
+function Headline:follow()
+  local headlines = Org.files:get_current_file():find_headlines_by_title(self.headline)
+
+  if #headlines == 0 then
+    return utils.echo_warning(('Could not find headline "%s".'):format(self.headline))
+  end
+
+  self.goto_oneof(headlines)
+end
+
+function Headline:resolve()
+  local headlines = Org.files:get_current_file():find_headlines_by_title(self.headline)
+
+  if #headlines == 0 then
+    return self
+  end
+
+  local id = headlines[1]:get_property('id')
```

Comment:
Why are we getting the id property here? Ids are handled through a separate handler.

Also, we need to consider that there are multiple headlines here.

Reply:
I think this was meant to implement the behaviour determined by `org-id-link-to-use-id` as described in https://orgmode.org/org.pdf#Handling%20Links.
Upon further reading however, this is resolved on storing a link, rather than on inserting it. I thought it happened at both. Removed